x-pack/plugins/siem/public/components/super_date_picker/index.tsx

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import dateMath from '@elastic/datemath';
+import {
+  EuiSuperDatePicker,
+  EuiSuperDatePickerProps,
+  OnRefreshChangeProps,
+  OnRefreshProps,
+  OnTimeChangeProps,
+} from '@elastic/eui';
+import { get, getOr } from 'lodash/fp';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { ActionCreator } from 'typescript-fsa';
+
+import { inputsActions, inputsModel, State } from '../../store';
+
+type MyEuiSuperDatePickerProps = Pick<
+  EuiSuperDatePickerProps,
+  | 'end'
+  | 'isPaused'
+  | 'onTimeChange'
+  | 'onRefreshChange'
+  | 'onRefresh'
+  | 'recentlyUsedRanges'
+  | 'refreshInterval'
+  | 'showUpdateButton'
+  | 'start'
+> & {
+  isLoading?: boolean;
+};
+const MyEuiSuperDatePicker: React.SFC<MyEuiSuperDatePickerProps> = EuiSuperDatePicker;
+
+interface SuperDatePickerStateRedux {
+  duration: number;
+  kind: string;
+  option: string;
+  start: number;
+  end: number;
+  isLoading: boolean;
+  refetch: inputsModel.Refetch[];
```

Comment:
the type of `Refetch` is a function:

```
export type Refetch = () => void;
```

which per the [Can I put functions, promises, or other non-serializable items in my store state? section of the Redux FAQ](https://redux.js.org/faq/organizing-state#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state) is _technically_ possible to store in Redux state, but discouraged.

Via inspection of the code / the Redux debug tools, it appears `refetch` is not actually part of the store (a good thing), but is being populated in `mapStateToProps`. If that's true, consider removing `refetch` from this interface, because `State` in the `interface` name `SuperDatePickerStateRedux` might imply to some readers that all of the props are coming from the Redux state, which may not be the case here.

Reply:
Here, I picked the redux store to keep track of my query to refetch. There is so many way that can be done. so if you feel really strongly about it, I can always manage it through a component or a context or whatever ;)